src/spec-node/devContainersSpecCLI.ts

```diff
@@ -872,8 +919,10 @@ export async function doExec({
 		if (!container) {
 			bailOut(common.output, 'Dev container not found.');
 		}
-		const containerProperties = await createContainerProperties(params, container.Id, workspaceConfig.workspaceFolder, config.remoteUser);
-		const updatedConfig = containerSubstitute(cliHost.platform, config.configFilePath, containerProperties.env, config);
+		const imageMetadata = getImageMetadataFromContainer(container, config, [], experimentalImageMetadata, output).config;
+		const mergedConfig = mergeConfiguration(config.config, imageMetadata);
```

Comment:
should it be `config` instead ?

Reply:
(Like above:) `config` has `config` and `raw` properties. I introduced an interface to carry the config with variables substituted and the raw config around:
```ts
interface SubstitutedConfig<T> {
	config: T;
	raw: T;
	substitute: SubstituteConfig;
}
```